feat: Use relative path for SwaggerUI

gaeljw opened the pull request:
Solves https://github.com/softwaremill/tapir/issues/2049

adamw commented:
Yeah the oauth2 integration needs to be fixed (see https://github.com/softwaremill/tapir/issues/1894). I assume the above works - as I mentioned would be best to include relative as an option (question is, should it be the default?). And add some tests :)

gaeljw commented:
I will work on it by next week :)

> question is, should it be the default?)

I would say yes, I believe it's the most common usage but really not a strong opinion on this.

gaeljw commented:
I had a few minutes to polish it.

I added tests in the bundle module as there were some already existing there and not in the swagger module. Not sure if you were expecting something else?

I also chose to set the relative path by default but really open to the other way around.

Feel free to suggest any change. Especially I'm not sure the piece of doc I updated is clear enough.

marcin-jozefowicz commented on docs/swagger-ui/src/main/scala/sttp/tapir/swagger/SwaggerUI.scala:
```diff
@@ -32,11 +32,13 @@ object SwaggerUI {
     val baseEndpoint = infallibleEndpoint.get.in(prefixInput)
     val redirectOutput = statusCode(StatusCode.PermanentRedirect).and(header[String](HeaderNames.Location))
 
+    val lastSegmentInput: EndpointInput[Option[String]] = extractFromRequest(request => request.pathSegments.lastOption)
```
Not sure if #extractFromRequest can be used : 
From docs: 
> /** Extract a value from a server request. This input is only used by server interpreters, it is ignored by documentation interpreters and
    * the provided value is discarded by client interpreters.
    */
    
 So looks like it will be skipped. @adamw Is it true?

adamw replied:
Well the result is a `ServerEndpoint` so it will be interpreted using a server interpreter, so it will work just fine :)

gaeljw replied:
Yup, I tested it on a real project and it does work.

gaeljw commented:
Thanks for the changes 👍

adamw commented:
I simplified things a bit - I think a reasonable assumption might be that if a `contextPath` is not specified, we should use the relative-redirect approach. If a context path is specified, we are using absolute addresses; plus, this automatically adds a `server` to the `OpenAPI` object. That way we avoid the possibility of creating invalid configuration (w/ `contextPath` specified and `useRelative = true`)

adamw commented:
Change my mind, reusing `contextPath` for this was only confusing ;) So I re-introduced the `useRelativePaths` config. Please take a look if everything looks ok:)

gaeljw commented:
Look's good, thanks again. Indeed using this automatically for the list of servers is slightly too "magical" IMHO.
